internal/impl/sftp/input.go

```diff
@@ -433,21 +448,22 @@ func (w *watcherPathProvider) Ack(ctx context.Context, name string, err error) (
 	return
 }
 
-func (s *sftpReader) getFilePathProvider(_ context.Context) pathProvider {
+func (s *sftpReader) getFilePathProvider(client *clientPool) pathProvider {
 	if !s.watcherEnabled {
-		var filepaths []string
-		for _, p := range s.paths {
-			paths, err := s.client.Glob(p)
+		provider := &staticPathProvider{}
+		for _, path := range s.paths {
+			expandedPaths, err := client.Glob(path)
 			if err != nil {
-				s.log.Warnf("Failed to scan files from path %v: %v", p, err)
+				s.log.Warnf("Failed to scan files from path %v: %v", path, err)
 				continue
 			}
-			filepaths = append(filepaths, paths...)
+			provider.expandedPaths = append(provider.expandedPaths, expandedPaths...)
```

Comment:
This `append` looks redundant since `provider.expandedPaths` is initialised on line 453